library/dhm.c

```diff
@@ -345,9 +345,6 @@ int mbedtls_dhm_make_public( mbedtls_dhm_context *ctx, int x_size,
     DHM_VALIDATE_RET( output != NULL );
     DHM_VALIDATE_RET( f_rng != NULL );
 
-    if( olen < 1 || olen > mbedtls_dhm_get_len( ctx ) )
-        return( MBEDTLS_ERR_DHM_BAD_INPUT_DATA );
-
     ret = dhm_make_common( ctx, x_size, f_rng, p_rng );
```

Comment:
Shouldn't we keep a minimum-length check of the form `if (olen < mbedtls_dhm_get_len(ctx))`?